control/dde.py

```diff
@@ -0,0 +1,374 @@
+        raise TypeError("Input must be a DelayLTI")
+
+    n_states = delay_sys.P.A.shape[0]
+    n_inputs = delay_sys.P.B1.shape[1]  # External inputs u
+    n_outputs = delay_sys.P.C1.shape[0]  # External outputs y
+
+    if U is not None:
+        U = np.asarray(U)
+    if T is not None:
+        T = np.asarray(T)
+
+    T = _check_convert_array(
+        T, [("any",), (1, "any")], "Parameter `T`: ", squeeze=True, transpose=transpose
+    )
+
+    n_steps = T.shape[0]
+    dt = (T[-1] - T[0]) / (n_steps - 1)
+    if not np.allclose(np.diff(T), dt):
+        raise ValueError("Parameter `T`: time values must be equally spaced.")
+
+    X0 = _check_convert_array(
+        X0, [(n_states,), (n_states, 1)], "Parameter `X0`: ", squeeze=True
+    )
+
+    # Test if U has correct shape and type
+    legal_shapes = (
+        [(n_steps,), (1, n_steps)] if n_inputs == 1 else [(n_inputs, n_steps)]
+    )
+    U = _check_convert_array(
+        U, legal_shapes, "Parameter `U`: ", squeeze=False, transpose=transpose
+    )
+    print("U shape: ", U.shape)
+    xout = np.zeros((n_states, n_steps))
+    xout[:, 0] = X0
+    yout = np.zeros((n_outputs, n_steps))
+    tout = T
+    xout, yout = solve_dde(delay_sys, T, U, X0, dt)
+
+    return TimeResponseData(
+        tout,
+        yout,
+        xout,
+        U,
+        params=params,
+        issiso=delay_sys.issiso(),
+        sysname=delay_sys.name,
+        plot_inputs=True,
+        title="Forced response for " + delay_sys.name,
+        trace_types=["forced"],
+        transpose=transpose,
+        return_x=return_x,
+        squeeze=squeeze,
+    )
+
+
+def pchip_interp_u(T, U):
+    """Create PCHIP interpolator functions for the input signal(s) U over time T.
+
+    For time points `t < T[0]`, the interpolator returns 0.
+
+    Parameters
+    ----------
+    T : array_like
+        Time vector, 1D array.
+    U : array_like
+        Input signal(s). Can be:
+        - 0D array (scalar): Assumed constant input. (Note: this path might
+          not be hit if U is pre-processed by `_check_convert_array`).
+        - 1D array `(n_steps,)`: Single input signal.
+        - 2D array `(n_inputs, n_steps)`: Multiple input signals.
+
+    Returns
+    -------
+    np.ndarray of PchipInterpolator or scalar
+        If U is 1D or 2D, returns a 1D NumPy array of PchipInterpolator
+        objects, one for each input signal.
+        If U is a scalar (0D), returns U itself.
+    """
+    def negative_wrapper(interp):
+        return lambda t: interp(t) if t >= T[0] else 0
+
+    if np.ndim(U) == 1:
+        # Single input signal, U.shape is (n_steps,)
+        return np.array([negative_wrapper(PchipInterpolator(T, U))])
+    elif np.ndim(U) == 0:
+        # Scalar input, assumed constant.
+        return U
+    else:
+        # Multiple input signals, U.shape is (n_inputs, n_steps)
+        return np.array([negative_wrapper(PchipInterpolator(T, ui)) for ui in U])
+
+
+class DdeHistory:
```

Comment:
This might be better named `DDEHistory` (since DDE is a standard abbreviation, like LTI)?

Also, if this is an internal class, begin the class name with an underscore.